tests/test_git_operations.py

```diff
@@ -0,0 +1,99 @@
+# MIT License
+#
+# Copyright (c) 2020 SCL team at Red Hat
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import pytest
+import shutil
+
+from pathlib import Path
+from flexmock import flexmock
+
+from container_workflow_tool.cli import ImageRebuilder
+
+from tests.spellbook import DATA_DIR
+
+
+class TestGitOperations(object):
+
+    def setup_method(self):
+        self.component = 's2i-base'
+        self.ir = ImageRebuilder('Testing')
+
+        self.ir.set_config('default.yaml', release="rawhide")
+        # Partner BZ testing
+        self.ir.rebuild_reason = "Unit testing"
+        self.ir.disable_klist = True
+        self.ir.set_do_images([self.component])
+
+
+    @pytest.mark.distgit
+    def test_distgit_commit_msg(self):
+        msg = "Unit testing"
+        self.ir.set_commit_msg(msg)
+        assert self.ir.distgit.commit_msg == msg
+
+    @pytest.mark.parametrize(
+        "os_name,os_name_expected,version",
+        [
+            ("fedora", "fedora", "base"),
+            ("RHEL8", "rhel8", "1.14"),
+            ("RHEL9", "rhel9", "1.14"),
+            ("RHSCL", "rhel7", "14"),
+            ("FOO", "fedora", "bar")
+        ]
+    )
+    def test_update_openshift_yaml(self, os_name, os_name_expected, version):
+        self.ir.conf.image_names = os_name
+        tmp = Path(self.ir._get_tmp_workdir())
```

Comment:
What about using pytest fixture for tmp directory so you can be sure that it will be deleted at the end?

Reply:
Good catch.

Reply:
Reworked.